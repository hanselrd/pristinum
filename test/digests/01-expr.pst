func main() @void:
    nil;
    true;
    false;
    'c';
    1;
    3.14159;
    "string";
    x;
    main();
    main(1);
    main(1, 2);
    (u64) 1;
    p x .``;
    p x ->``;
    `*p;
    `&p;
    size? u64;
    align? u64;
    x ++`;
    x --`;
    x -`;
    x ~`;
    x !`;
    x y +``;
    x y -``;
    x y *``;
    x y /``;
    x y %``;
    x y **``;
    x y <<``;
    x y >>``;
    x y &``;
    x y |``;
    x y ^``;
    x y ==``;
    x y !=``;
    x y <``;
    x y >``;
    x y <=``;
    x y >=``;
    x y &&``;
    x y ||``;
end