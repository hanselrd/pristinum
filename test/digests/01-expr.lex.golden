[ PT
    ( Pn 0 1 1 )
    ( TK
        ( TokSymbol
            { tsText = "func"
            , tsID = 42
            }
        )
    )
, PT
    ( Pn 5 1 6 )
    ( T_IDENT "main" )
, PT
    ( Pn 9 1 10 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 10 1 11 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 12 1 13 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 13 1 14 )
    ( TK
        ( TokSymbol
            { tsText = "void"
            , tsID = 51
            }
        )
    )
, PT
    ( Pn 17 1 18 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 23 2 5 )
    ( TK
        ( TokSymbol
            { tsText = "nil"
            , tsID = 45
            }
        )
    )
, PT
    ( Pn 26 2 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 32 3 5 )
    ( TK
        ( TokSymbol
            { tsText = "true"
            , tsID = 49
            }
        )
    )
, PT
    ( Pn 36 3 9 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 42 4 5 )
    ( TK
        ( TokSymbol
            { tsText = "false"
            , tsID = 41
            }
        )
    )
, PT
    ( Pn 47 4 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 53 5 5 )
    ( TC "'c'" )
, PT
    ( Pn 56 5 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 62 6 5 )
    ( TI "1" )
, PT
    ( Pn 63 6 6 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 69 7 5 )
    ( TD "3.14159" )
, PT
    ( Pn 76 7 12 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 82 8 5 )
    ( TL "string" )
, PT
    ( Pn 90 8 13 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 96 9 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 97 9 6 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 103 10 5 )
    ( T_IDENT "main" )
, PT
    ( Pn 107 10 9 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 108 10 10 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 109 10 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 115 11 5 )
    ( T_IDENT "main" )
, PT
    ( Pn 119 11 9 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 120 11 10 )
    ( TI "1" )
, PT
    ( Pn 121 11 11 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 122 11 12 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 128 12 5 )
    ( T_IDENT "main" )
, PT
    ( Pn 132 12 9 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 133 12 10 )
    ( TI "1" )
, PT
    ( Pn 134 12 11 )
    ( TK
        ( TokSymbol
            { tsText = ","
            , tsID = 12
            }
        )
    )
, PT
    ( Pn 136 12 13 )
    ( TI "2" )
, PT
    ( Pn 137 12 14 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 138 12 15 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 144 13 5 )
    ( TK
        ( TokSymbol
            { tsText = "cast"
            , tsID = 35
            }
        )
    )
, PT
    ( Pn 148 13 9 )
    ( TK
        ( TokSymbol
            { tsText = "<"
            , tsID = 21
            }
        )
    )
, PT
    ( Pn 149 13 10 )
    ( T_IDENT "u64" )
, PT
    ( Pn 152 13 13 )
    ( TK
        ( TokSymbol
            { tsText = ">"
            , tsID = 26
            }
        )
    )
, PT
    ( Pn 153 13 14 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 154 13 15 )
    ( TI "1" )
, PT
    ( Pn 155 13 16 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 156 13 17 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 162 14 5 )
    ( T_IDENT "p" )
, PT
    ( Pn 163 14 6 )
    ( TK
        ( TokSymbol
            { tsText = "."
            , tsID = 16
            }
        )
    )
, PT
    ( Pn 164 14 7 )
    ( T_IDENT "x" )
, PT
    ( Pn 165 14 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 171 15 5 )
    ( T_IDENT "p" )
, PT
    ( Pn 172 15 6 )
    ( TK
        ( TokSymbol
            { tsText = "."
            , tsID = 16
            }
        )
    )
, PT
    ( Pn 173 15 7 )
    ( T_IDENT "x" )
, PT
    ( Pn 174 15 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 180 16 5 )
    ( TK
        ( TokSymbol
            { tsText = "*"
            , tsID = 8
            }
        )
    )
, PT
    ( Pn 181 16 6 )
    ( T_IDENT "p" )
, PT
    ( Pn 182 16 7 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 188 17 5 )
    ( TK
        ( TokSymbol
            { tsText = "&"
            , tsID = 4
            }
        )
    )
, PT
    ( Pn 189 17 6 )
    ( T_IDENT "p" )
, PT
    ( Pn 190 17 7 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 196 18 5 )
    ( TK
        ( TokSymbol
            { tsText = "sizeof"
            , tsID = 47
            }
        )
    )
, PT
    ( Pn 202 18 11 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 203 18 12 )
    ( T_IDENT "u64" )
, PT
    ( Pn 206 18 15 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 207 18 16 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 213 19 5 )
    ( TK
        ( TokSymbol
            { tsText = "alignof"
            , tsID = 33
            }
        )
    )
, PT
    ( Pn 220 19 12 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 221 19 13 )
    ( T_IDENT "u64" )
, PT
    ( Pn 224 19 16 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 225 19 17 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 231 20 5 )
    ( TK
        ( TokSymbol
            { tsText = "++"
            , tsID = 11
            }
        )
    )
, PT
    ( Pn 233 20 7 )
    ( T_IDENT "x" )
, PT
    ( Pn 234 20 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 240 21 5 )
    ( TK
        ( TokSymbol
            { tsText = "--"
            , tsID = 14
            }
        )
    )
, PT
    ( Pn 242 21 7 )
    ( T_IDENT "x" )
, PT
    ( Pn 243 21 8 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 249 22 5 )
    ( TK
        ( TokSymbol
            { tsText = "-"
            , tsID = 13
            }
        )
    )
, PT
    ( Pn 250 22 6 )
    ( T_IDENT "x" )
, PT
    ( Pn 251 22 7 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 257 23 5 )
    ( TK
        ( TokSymbol
            { tsText = "~"
            , tsID = 55
            }
        )
    )
, PT
    ( Pn 258 23 6 )
    ( T_IDENT "x" )
, PT
    ( Pn 259 23 7 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 265 24 5 )
    ( TK
        ( TokSymbol
            { tsText = "!"
            , tsID = 1
            }
        )
    )
, PT
    ( Pn 266 24 6 )
    ( T_IDENT "x" )
, PT
    ( Pn 267 24 7 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 273 25 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 275 25 7 )
    ( TK
        ( TokSymbol
            { tsText = "+"
            , tsID = 10
            }
        )
    )
, PT
    ( Pn 277 25 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 278 25 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 284 26 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 286 26 7 )
    ( TK
        ( TokSymbol
            { tsText = "-"
            , tsID = 13
            }
        )
    )
, PT
    ( Pn 288 26 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 289 26 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 295 27 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 297 27 7 )
    ( TK
        ( TokSymbol
            { tsText = "*"
            , tsID = 8
            }
        )
    )
, PT
    ( Pn 299 27 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 300 27 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 306 28 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 308 28 7 )
    ( TK
        ( TokSymbol
            { tsText = "/"
            , tsID = 17
            }
        )
    )
, PT
    ( Pn 310 28 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 311 28 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 317 29 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 319 29 7 )
    ( TK
        ( TokSymbol
            { tsText = "%"
            , tsID = 3
            }
        )
    )
, PT
    ( Pn 321 29 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 322 29 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 328 30 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 330 30 7 )
    ( TK
        ( TokSymbol
            { tsText = "**"
            , tsID = 9
            }
        )
    )
, PT
    ( Pn 333 30 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 334 30 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 340 31 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 342 31 7 )
    ( TK
        ( TokSymbol
            { tsText = "<<"
            , tsID = 22
            }
        )
    )
, PT
    ( Pn 345 31 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 346 31 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 352 32 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 354 32 7 )
    ( TK
        ( TokSymbol
            { tsText = ">>"
            , tsID = 28
            }
        )
    )
, PT
    ( Pn 357 32 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 358 32 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 364 33 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 366 33 7 )
    ( TK
        ( TokSymbol
            { tsText = "&"
            , tsID = 4
            }
        )
    )
, PT
    ( Pn 368 33 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 369 33 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 375 34 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 377 34 7 )
    ( TK
        ( TokSymbol
            { tsText = "|"
            , tsID = 53
            }
        )
    )
, PT
    ( Pn 379 34 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 380 34 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 386 35 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 388 35 7 )
    ( TK
        ( TokSymbol
            { tsText = "^"
            , tsID = 32
            }
        )
    )
, PT
    ( Pn 390 35 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 391 35 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 397 36 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 399 36 7 )
    ( TK
        ( TokSymbol
            { tsText = "=="
            , tsID = 25
            }
        )
    )
, PT
    ( Pn 402 36 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 403 36 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 409 37 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 411 37 7 )
    ( TK
        ( TokSymbol
            { tsText = "!="
            , tsID = 2
            }
        )
    )
, PT
    ( Pn 414 37 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 415 37 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 421 38 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 423 38 7 )
    ( TK
        ( TokSymbol
            { tsText = "<"
            , tsID = 21
            }
        )
    )
, PT
    ( Pn 425 38 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 426 38 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 432 39 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 434 39 7 )
    ( TK
        ( TokSymbol
            { tsText = ">"
            , tsID = 26
            }
        )
    )
, PT
    ( Pn 436 39 9 )
    ( T_IDENT "y" )
, PT
    ( Pn 437 39 10 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 443 40 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 445 40 7 )
    ( TK
        ( TokSymbol
            { tsText = "<="
            , tsID = 23
            }
        )
    )
, PT
    ( Pn 448 40 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 449 40 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 455 41 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 457 41 7 )
    ( TK
        ( TokSymbol
            { tsText = ">="
            , tsID = 27
            }
        )
    )
, PT
    ( Pn 460 41 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 461 41 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 467 42 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 469 42 7 )
    ( TK
        ( TokSymbol
            { tsText = "&&"
            , tsID = 5
            }
        )
    )
, PT
    ( Pn 472 42 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 473 42 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 479 43 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 481 43 7 )
    ( TK
        ( TokSymbol
            { tsText = "||"
            , tsID = 54
            }
        )
    )
, PT
    ( Pn 484 43 10 )
    ( T_IDENT "y" )
, PT
    ( Pn 485 43 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 487 44 1 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 39
            }
        )
    )
]